process/src/process.cpp

```diff
@@ -0,0 +1,398 @@
+#include "lute/process.h"
+#include "lute/runtime.h"
+#include <uv.h>
+#include <string>
+#include <vector>
+#include <memory>
+#include <functional>
+#include <map>
+
+#include "lua.h"
+#include "lualib.h"
+
+namespace process
+{
+
+struct ProcessHandle
+{
+    uv_process_t process;
+    uv_pipe_t stdout_pipe;
+    uv_pipe_t stderr_pipe;
+    uv_loop_t* loop = nullptr;
+    std::string stdout_data;
+    std::string stderr_data;
+    int64_t exit_code = -1;
+    int term_signal = 0;
+    bool completed = false;
+    ResumeToken resume_token;
+    std::shared_ptr<ProcessHandle> self;
+    std::atomic<int> pending_closes{0};
+
+    ~ProcessHandle() {}
+
+    void close_handles()
```

Comment:
Annoying nitpicky stuff, but all internal stuff should be camelCase, only the exposed API is snake_case. This is carrying over the same C++ style as Luau itself.

Reply:
All good, I totally forgot about this, fixed it all up I think